src/Microsoft.DotNet.Compiler.Common/ProjectContextExtensions.cs

```diff
@@ -37,6 +37,12 @@ public static string GetSDKVersionFile(this ProjectContext context, string confi
             return Path.Combine(intermediatePath, ".SDKVersion");
         }
 
+        public static string IncrementalCacheFile(this ProjectContext context, string configuration, string buildBasePath, string outputPath)
+        {
+            var intermediatePath = context.GetOutputPaths(configuration, buildBasePath, outputPath).IntermediateOutputDirectoryPath;
+            return Path.Combine(intermediatePath, ".IncrementalCache");
```

Comment:
We could all the file names from the intermediary output directory somewhere in utils, or in the common compiler (so there's no cyclic dependencies).

This would be the names for:
- sdk version file,
- incremental cache,
- response files (compile.rsp, compile-csc.rsp)
etc